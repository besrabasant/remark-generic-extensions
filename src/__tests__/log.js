import test from "ava"

import { transformToLog } from "./helpers/transform"

test(
  "should show a `no issues found` message",
  t => t.snapshot(transformToLog("!icon"))
)

test(
  "should log information if debug is enabled",
  t => t.snapshot(transformToLog("!icon", { debug: true }))
)

test(
  "should log a warning if there is an invalid property",
  t => t.snapshot(transformToLog("!icon{/foo}"))
)
