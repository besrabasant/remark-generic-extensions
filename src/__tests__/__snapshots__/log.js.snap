// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should log a warning if there is an invalid property 1`] = `
"  1:1  [33mwarning[39m  There was some invalid properties: \\"/\\" was left after the processing of \\"/foo\\"

[33mâš [39m 1 warning"
`;

exports[`should log information if debug is enabled 1`] = `
"  1:1  [33mwarning[39m  Inline extension \`icon\` found  debug
  1:1  [33mwarning[39m  Full match: \\"!icon\\"            debug
  1:1  [33mwarning[39m  Content: \\"\\"                    debug
  1:1  [33mwarning[39m  Argument: \\"\\"                   debug
  1:1  [33mwarning[39m  Properties string: \\"\\"          debug
  1:1  [33mwarning[39m  Computed properties: {}        debug
  1:1  [33mwarning[39m  Hast output tree:
{
  \\"type\\": \\"inline-extension\\",
  \\"data\\": {
    \\"hName\\": \\"icon\\",
    \\"hChildren\\": [],
    \\"hProperties\\": {}
  }
}              debug

[33mâš [39m 7 warnings"
`;

exports[`should show a \`no issues found\` message 1`] = `"no issues found"`;
